src/pages/FindDoctors/FindDoctors.tsx

```diff
@@ -6,54 +6,72 @@ import { CardDoctor } from "../../components/PageDoctors/CardDoctor";
 import { getDoctors } from "../../api/doctors";
 import { useNavigate, useSearchParams } from "react-router-dom";
 import { YMaps, Map, Placemark } from "@pbe/react-yandex-maps";
-import { useQuery } from "react-query";
-import { CircularProgress, Grid } from "@mui/material";
-
-const defaultState = {
-  center: [55.751574, 37.573856],
-  zoom: 10,
-};
+import { Grid } from "@mui/material";
+import { Search } from "../../components/SectionFindDoctors/Search/Search";
 
 const FindDoctors = () => {
-  const navigate = useNavigate();
   const [coordsData, setCoordsData] = useState<[number, number][]>([]);
+  const [data, setData] = useState<IDoctor[]>([]);
+  const [defaultState, setDefaultState] = useState({
+    center: [55.751574, 37.573856],
+    zoom: 10,
+  });
   const [searchParams] = useSearchParams();
+  const navigate = useNavigate();
 
-  const city = searchParams.get("city") || "";
-  const specialization = searchParams.get("specialization") || "";
-  console.log(city, specialization);
-
-  const { isLoading, data } = useQuery("doctors", () => getDoctors(specialization, city));
+  // const { isLoading, data } = useQuery("doctors", () => getDoctors(specialization, city));
 
   useEffect(() => {
-    async function getCoords(data: IDoctor[]) {
-      if (data == undefined) return;
+    console.log("AAAAAAAAAAAAAAAAAAAAAAAAAAAAAa");
+
+    const city = searchParams.get("city") || "";
+    const specialization = searchParams.get("specialization") || "";
+
+    async function getCoords() {
```

Comment:
This function is absolutely not correct.
First of all, u can use use react query instead of combination of useEffect with useState.
Secondly, `data.toString() !== dataDoc.toString() ? setData(dataDoc) : true;` looks really unnecessary, as state will be the same.
Algorithm is the following:
1) U are fetching doctor's data using react query (with filtering docs without address)
2) Using dependent query https://react-query-v3.tanstack.com/guides/dependent-queries#_top u are fetching map data by address.
3) Using onSuccess method https://react-query-v3.tanstack.com/reference/useQuery#_top u are setting coordinates and default state.